tooling/ipns/ipns.go

```diff
@@ -0,0 +1,59 @@
+package ipns
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+
+	"github.com/ipfs/gateway-conformance/tooling/fixtures"
+)
+
+/**
+ * Extracts the public key from the path of an IPNS record.
+ * The path is expected to be in the format of:
+ * some/path/then/[pubkey](_anything)?.ipns-record
```

Comment:
Dear reviewer, one highlight for this PR:
we need to move the pubkey around with the ipns-record, we keep it in the filename, which gets the job done for now, but is not the greatest solution (ideally we'll store them in the car encoding)

Reply:
:+1:  I think this is a very sensible convention until we have https://github.com/ipfs/specs/issues/369 figured  out.